services/pull/merge.go

```diff
@@ -470,6 +470,23 @@ func getDiffTree(repoPath, baseBranch, headBranch string) (string, error) {
 	return out.String(), nil
 }
 
+// IsSignedIfRequired check if merge will be signed if required
+func IsSignedIfRequired(pr *models.PullRequest, doer *models.User) (bool, error) {
+	if pr.ProtectedBranch == nil {
```

Comment:
May be it's better add `if pr.ProtectedBranch == nil` to `(pr *PullRequest) loadProtectedBranch()` rather than here?

Reply:
OK. I just didn't want to assume that this would not cause NPEs down the line

Reply:
Seems to pass the build so I've done that.